cluster_i_test.go

```diff
@@ -380,3 +377,75 @@ func TestEnqueueCommandsAndRetryFromQueue(t *testing.T) {
 		}
 	}
 }
+
+func TestRecoverFromReadTimeout(t *testing.T) {
```

Comment:
Note: the "test" is the fact that this executes without error. Before fixing the bugs in the code, this test would either panic or error out.